index.js

```diff
@@ -46,12 +47,19 @@ async function sendRewards(targets) {
     const tx = await distributor.send(addresses, amounts)
     const tr = await tx.wait()
     console.log("Gas spent: %s", tr.gasUsed.toString())
+
+    // persist the last rewarded index for automatic recovery
+    if (STATE_FILE_NAME) {
+        await fs.writeFile(STATE_FILE_NAME, last.index)
+    }
 }
 
 async function main() {
     console.log("Connected to network %o", await provider.getNetwork())
 
-    const rawInput = (await fs.readFile(INPUT, "utf8")).split("\n").slice(START, END)
+    const lastRewardedIndex = STATE_FILE_NAME && fs.existsSync(STATE_FILE_NAME) ? +fs.readFileSync(STATE_FILE_NAME, "utf8") : -1
+    const startIndex = START === "" ? +START : lastRewardedIndex + 1
```

Comment:
If `START` is empty, then use it? I'm wondering if you meant

```suggestion
    const startIndex = START !== "" ? +START : lastRewardedIndex + 1
```

Reply:
👍